vault/resource_ad_secret_backend_test.go

```diff
@@ -74,6 +74,49 @@ func TestADSecretBackend(t *testing.T) {
 	})
 }
 
+func TestADSecretBackend_remount(t *testing.T) {
+	backend := acctest.RandomWithPrefix("tf-test-ad")
+	updatedBackend := acctest.RandomWithPrefix("tf-test-ad-updated")
+
+	resourceName := "vault_ad_secret_backend.test"
+	bindDN, bindPass, url := testutil.GetTestADCreds(t)
+
+	resource.Test(t, resource.TestCase{
+		Providers: testProviders,
+		PreCheck:  func() { testutil.TestAccPreCheck(t) },
+		Steps: []resource.TestStep{
+			{
+				Config: testADSecretBackend_initialConfig(backend, bindDN, bindPass, url),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "backend", backend),
+					resource.TestCheckResourceAttr(resourceName, "description", "test description"),
+					resource.TestCheckResourceAttr(resourceName, "default_lease_ttl_seconds", "3600"),
+					resource.TestCheckResourceAttr(resourceName, "max_lease_ttl_seconds", "7200"),
+					resource.TestCheckResourceAttr(resourceName, "binddn", bindDN),
+					resource.TestCheckResourceAttr(resourceName, "bindpass", bindPass),
+					resource.TestCheckResourceAttr(resourceName, "url", url),
+					resource.TestCheckResourceAttr(resourceName, "insecure_tls", "true"),
+					resource.TestCheckResourceAttr(resourceName, "userdn", "CN=Users,DC=corp,DC=example,DC=net"),
+				),
+			},
```

Comment:
Should we be testing import?

Reply:
Added import tests for all new tests for secret/auth backends